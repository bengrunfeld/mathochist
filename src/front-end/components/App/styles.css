h1 {
  color: darkblue;
}

.selfie {
  background: url(../../img/smiley.jpg);
  background-size: 150px;
  background-repeat-y: no-repeat;
  width: 150px;
  height: 128px;
}