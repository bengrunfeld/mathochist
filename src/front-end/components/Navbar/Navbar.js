import React from 'react'

const Navbar = () => (
  <div>
    <p>Navbar</p>
  </div>
)

export default Navbar