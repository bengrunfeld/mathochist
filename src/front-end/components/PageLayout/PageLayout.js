import React, { useState } from 'react'

import Sidebar from '../Sidebar'
import Page from '../Page'

const PageLayout = () => {
  const [state, setState] = useState({postId: ''})

  return (
    <div className='container-fluid'>
      <div className='row'>
        <Sidebar updateState={setState} />
        <Page postId={state.postId} />
      </div>
    </div>
  )
}

export default PageLayout
