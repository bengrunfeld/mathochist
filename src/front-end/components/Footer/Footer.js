import React from 'react'

import { FooterBar } from './styles'

const Footer = () => (
  <div className='container-fluid'>
    <FooterBar className='row'>
      <div className='col-12'>
        <h2>Footer</h2>
        <p>Other text</p>
      </div>
    </FooterBar>
  </div>
)

export default Footer
