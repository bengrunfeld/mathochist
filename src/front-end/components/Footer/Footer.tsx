import React from 'react'

import { FooterBar } from './styles'

const Footer = () => (
  <div className='container-fluid'>
    <FooterBar className='row'>
      <div className='col-6'>
        <p>Â© Ben Grunfeld 2019</p>
      </div>
    </FooterBar>
  </div>
)

export default Footer
