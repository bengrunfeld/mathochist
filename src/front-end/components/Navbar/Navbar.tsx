import React from 'react'

import { NavbarContainer, Logo } from './styles'

const Navbar = () => (
  <NavbarContainer className='navbar navbar-expand-lg'>
    <Logo className='navbar-brand ml-2' to='/'>
      Mathochist
    </Logo>
  </NavbarContainer>
)

export default Navbar
