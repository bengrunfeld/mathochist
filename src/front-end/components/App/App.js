import React from 'react'
import Navbar from '../Navbar'
import PageLayout from '../PageLayout'
import Footer from '../Footer'

import './styles.css'

import 'bootstrap'

import '../../css/resets.css'
import '../../css/fonts.css'
import '../../../../node_modules/bootstrap/dist/css/bootstrap.min.css'


const App = () => (
  <div>
    <Navbar />
    <PageLayout />
    <Footer />
  </div>
)

export default App