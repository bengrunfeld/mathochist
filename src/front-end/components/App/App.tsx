import React from 'react'
import Navbar from '../Navbar'
import PageLayout from '../PageLayout'
import Footer from '../Footer'
import { Map } from 'immutable'

import { connect } from 'react-redux'

import './styles.css'

import 'bootstrap'

import '../../assets/css/resets.css'
import '../../assets/css/fonts.css'
import '../../assets/css/typography.css'
import '../../../../node_modules/bootstrap/dist/css/bootstrap.min.css'

const App = ({ time, goals, points }) => (
  <div>
    <h3>The time is: {time}</h3>
    <p>
      Score is: {goals} goals and {points} points
    </p>
    <Navbar />
    <PageLayout />
    <Footer />
  </div>
)

const mapStateToProps = state => ({
  time: state.time,
  goals: state.goals,
  points: state.points
})

const mapDispatchToProps = {
  fetchData: 'hello'
}

export default connect(mapStateToProps)(App)
