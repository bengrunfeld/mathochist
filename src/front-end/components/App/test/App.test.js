import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import App from '../App'
import Navbar from '../../Navbar'
import PageLayout from '../../PageLayout'
import Footer from '../../Footer'

import setUpTree from '../../../../test/setUpTree'


describe('<App />', () => {
  test('renders a Navbar', () => {
    const tree = setUpTree(App, MemoryRouter, mount, '/')
    expect(tree.find(Navbar)).toHaveLength(1)
  })

  test('renders a PageLayout', () => {
    const tree = setUpTree(App, MemoryRouter, mount, '/')
    expect(tree.find(PageLayout)).toHaveLength(1)
  })

  test('renders a Footer', () => {
    const tree = setUpTree(App, MemoryRouter, mount, '/')
    expect(tree.find(Footer)).toHaveLength(1)
  })

  test('matches the snapshot', () => {
    const tree = setUpTree(App, MemoryRouter, mount, '/')
    expect(tree).toMatchSnapshot()
  })
})
