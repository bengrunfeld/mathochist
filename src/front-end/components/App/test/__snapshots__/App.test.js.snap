// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App>
      <h1>
        My name is 
         and I am 
        NaN
         years old.
      </h1>
    </App>
  </Router>
</MemoryRouter>
`;
